addons/csw/functions/fnc_assemble_pickupWeapon.sqf

```diff
@@ -0,0 +1,82 @@
+ *
+ * Arguments:
+ * 0: Static Weapon <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [weapon] call ace_csw_fnc_assemble_pickupWeapon
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+[{
+    params ["_staticWeapon", "_player"];
+    TRACE_2("assemble_pickupWeapon",_staticWeapon,_player);
+
+    private _carryWeaponClassname = getText(configFile >> "CfgVehicles" >> (typeOf _staticWeapon) >> QUOTE(ADDON) >> "disassembleWeapon");
+    private _turretClassname = getText(configFile >> "CfgVehicles" >> (typeOf _staticWeapon) >> QUOTE(ADDON) >> "disassembleTurret");
+    private _pickupTime = getNumber(configFile >> "CfgWeapons" >> _carryWeaponClassname >> QUOTE(ADDON) >> "pickupTime");
+    TRACE_4("",typeOf _staticWeapon,_carryWeaponClassname,_turretClassname,_pickupTime);
+    if (!isClass (configFile >> "CfgWeapons" >> _carryWeaponClassname)) exitWith {ERROR_1("bad weapon classname [%1]",_carryWeaponClassname);};
+    if (!isClass (configFile >> "CfgVehicles" >> _turretClassname)) exitWith {ERROR_1("bad turret classname [%1]",_turretClassname);};
+
+    private _onFinish = {
+        params ["_args"];
+        _args params ["_staticWeapon", "_player", "_carryWeaponClassname", "_turretClassname"];
+        TRACE_4("disassemble finish",_staticWeapon,_player,_carryWeaponClassname,_turretClassname);
+
+        private _weaponPos = getPosATL _staticWeapon;
+        _weaponPos set [2, (_weaponPos select 2) + 0.1];
+        private _weaponDir = getDir _staticWeapon;
+
+        LOG("remove ammo");
+        {
+            _x params ["_xMag", "", "_xAmmo"];
+
+            private _carryMag = GVAR(vehicleMagCache) getVariable _xMag;
+            if (isNil "_carryMag") then {
+                private _groups = "getNumber (_x >> _xMag) == 1" configClasses (configFile >> QGVAR(groups));
+                _carryMag = configName (_groups param [0, configNull]);
+                GVAR(vehicleMagCache) setVariable [_xMag, _carryMag];
+                TRACE_2("setting cache",_xMag,_carryMag);
+            };
+            if ((_xAmmo > 0) && {_carryMag != ""}) then {
+                TRACE_2("Removing ammo",_xMag,_carryMag);
+                [_player, _carryMag, _xAmmo] call FUNC(reload_handleReturnAmmo);
+            };
+        } forEach (magazinesAllTurrets _staticWeapon);
+
+        LOG("delete weapon");
+
+        deleteVehicle _staticWeapon;
+
+        private _cswTripod = createVehicle [_turretClassname, [0, 0, 0], [], 0, "NONE"];
+        _cswTripod setPosATL _weaponPos;
+        _cswTripod setDir _weaponDir;
+        _cswTripod setVelocity [0, 0, -0.05];
+        _cswTripod setVectorUp (surfaceNormal _weaponPos);
+
+        _weaponRelPos = _cswTripod getRelPos RELATIVE_DIRECTION(90);
+        private _weaponHolder = createVehicle ["groundWeaponHolder", [0, 0, 0], [], 0, "NONE"];
+        _weaponHolder setPosATL [_weaponRelPos select 0, _weaponRelPos select 1, _weaponPos select 2];
+        _weaponHolder setDir random [0, 180, 360];
```

Comment:
I found weird behaviour of this command on some islands. Weaponholder spawn with negative Z ATL coordinates.
Quote from BIKI:

> Make sure you setDir BEFORE you set position. Setting direction after set position could lead to unpredictable behaviour. For example main part of the hospital building in Arma 3 can lose collision detection near both side entrances. AI will also get confused and will stop detecting obstacles if setDir is called after setPos.

So I suppose it should be swapped